frontend/src/utils.ts

```diff
@@ -23,11 +23,54 @@ export function userProfileUrl(username: string): string {
    return `https://github.com/${safeUsername}`;
 }
 
-export const dummyPulls: PullData[] = (process.env.DUMMY_PULLS || []) as PullData[];
+export const dummyPulls: PullData[] = fakeCIStatuses((process.env.DUMMY_PULLS || []) as PullData[]);
 export function hasDummyPulls() {
    return dummyPulls.length > 0;
 }
 
 export function newTab(url: string) {
    window.open(url, "_blank");
 }
+
+function fakeCIStatuses(pulls: PullData[]) {
+   const fakeContexts = ['psalm', 'deploy', 'unit-tests', 'api', 'integration', 'browser', 'bundle-analysis'];
+   pulls.forEach((pull) => {
+      pull.status.commit_statuses = fakeContexts.map((context) => {
+         const now = unix();
+         const startedAt = now - (Math.random() * 120 * 60);
+         const duration = Math.random() * 10 * 60;
+         const completedAt = startedAt + duration;
+         return {data: {
+            sha: "fake sha",
+            target_url: "https://example.com",
+            description: context,
+            context: context,
+            state: completedAt > now ? StatusState.pending : finishedState(),
+            started_at: startedAt,
+            completed_at: completedAt > now ? null : completedAt,
+         }};
+      });
+   });
+   return pulls;
+}
+
+function finishedState() {
+   const r = Math.random();
+   return r < 0.01 ? StatusState.error :
+      (r < 0.03 ? StatusState.failure : StatusState.success);
```

Comment:
Just to be clear, I think this is
1% of the time, return `StatusState.error`
2% of the time, return `StatusState.failure`
97% of the time, return `StatusState.success`

I'm suspicious that the 2% there is supposed to be 3%; I believe it's only 2% because
`0 <= r < 0.01` => `StatusState.error`
and thus
`0.01 <= r < 0.03` => `StatusState.failure`, which is `0.03-0.01 = 0.02 = 2%`.

Reply:
Yes, you are likely right that I was moving fast and didn't think about the 3 - 1 == 2, but I'm OK with that. This isn't critical, it's just to give a sprinkling of different states.